internal/contracts/manager.go

```diff
@@ -877,26 +1022,92 @@ func (cm *contractManager) AddContractListener(ctx context.Context, listener *co
 		} else if len(existing) > 0 {
 			return i18n.NewError(ctx, coremsgs.MsgContractListenerExists)
 		}
+
+		// Check for existense of an older listener with the old signature
+		if listener.Event != nil {
+			// Note the event signature has been extended with more information in some blockchain plugins
+			// That is why we do not add the signature in the query but instead iterate over the listeners
+			// and compare the signatures
+			signature := cm.blockchain.GenerateEventSignature(ctx, &listener.Event.FFIEventDefinition)
+			filter := database.ContractListenerQueryFactory.NewFilter(ctx)
+			if existing, _, err := cm.database.GetContractListeners(ctx, cm.namespace, filter.And(
+				filter.Eq("topic", listener.Topic),
+				filter.Eq("location", locationLookup),
+			)); err != nil {
+				return err
+			} else if len(existing) > 0 {
+				for _, listener := range existing {
+					// We have extended the event signature to add more information
+					// So we compare the start with is not guaranteed to be the same
+					// but it's the best comparison
+					if strings.HasPrefix(signature, listener.Signature) {
+						return i18n.NewError(ctx, coremsgs.MsgContractListenerExists)
```

Comment:
Again maybe this comparing of signatures should be plugin specific

Reply:
I don't think comparing signatures should be specific, but I don't understand the `HasPrefix` here.
Surely the signatures are the same, or they are not.